src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -74,6 +74,16 @@ class IAggregateFunction : public std::enable_shared_from_this<IAggregateFunctio
     /// Get the data type of internal state. By default it is AggregateFunction(name(params), argument_types...).
     virtual DataTypePtr getStateType() const;
 
+    /// Returns true if two aggregate functions have the same state representation in memory and the same serialization,
+    /// so state of one aggregate function can be safely used with another.
+    /// Examples:
+    ///  - quantile(x), quantile(a)(x), quantile(b)(x) - parameter doesn't affect state and used for finalization only
+    ///  - foo(x) and fooIf(x) - If combinator doesn't affect state
```

Comment:
You don't implement this case (for -If) in this PR, right?

Reply:
Yes, it will not work anyway right now and require more complex changes (`-If` combinator adds one extra argument and there is another place where number of arguments is checked)

Reply:
OK. The extra argument doesn't influence the serialization, so probably we can fall through to the nested->hasSameStateRepresentation.